### ISSUE TITLE
Static portion of templates are changing?

### ISSUE THREAD

**Issue by gkuchta** [hashicorp/consul-template#222]
Hi,

This is sort of blowing my mind. I have the following template defined:

```
global
  log 127.0.0.1 local0

defaults REDIS
 log global
 mode tcp
 timeout connect  4s
 timeout server  30s
 timeout client  30s

frontend redis_frontend
 bind 192.168.217.145:6379  name redis
 default_backend redis_backend

backend redis_backend
 option tcp-check
 tcp-check send PING\r\n
 tcp-check expect string +PONG
 tcp-check send info\ replication\r\n
 tcp-check expect string role:master
 tcp-check send QUIT\r\n
 tcp-check expect string +OK

{{range service "radar-redis"}}
server {{.Name}} {{.Address}}:{{.Port}} check inter 1s
{{end}}
```

I'm not sure how, but this is what's generated:

```
global
  log 127.0.0.1 local0

defaults REDIS
 log global
 mode tcp
 timeout connect  4s
 timeout server  30s
 timeout client  30s

frontend redis_frontend
 bind 127.0.0.1:6379  name redis
 default_backend redis_backend

backend redis_backend
 option tcp-check
 tcp-check send PING\r\n
 tcp-check expect string +PONG
 tcp-check send info\ replication\r\n
 tcp-check expect string role:master
 tcp-check send QUIT\r\n
 tcp-check expect string +OK


server radar-redis 192.168.217.147:16379 check inter 1s
```

Somehow the bind address is changing to 127.0.0.1; I don't have chef or anything else trying to keep an eye on that file. I'm not sure what debugging info I can provide that'd be useful here; strace?

**Comment by hoffoo** [comment]
maybe running consul-template with -log-level=debug would be useful

**Comment by ryanuber** [comment]
@gkuchta is it possible that you have a service restart command attached to the template? What I'm thinking is that the service itself could be updating the value during start. If you do have a restart command or some other command configured, could you try starting consul-template without it and see if it produces the same result?

**Comment by sielaq** [comment]
@hoffoo , Do you run "template for a template", kind of dynamic configuration for config-template ?

**Comment by sethvargo** [comment]
Hey @gkuchta 

I am sorry Consul Template is causing you issues. Could you please run the command in debug + dry mode and gist the output for us?

**Comment by gkuchta** [comment]
Hi,

After a lot of defeated stares/some existential soul searching, it looks like I may have tracked the issue down: subtle and very bizarre file system corruption. The VM image I was building upon had been snapshotted without having its' disks sync'd or buffers flushed -- I noticed this when I saw a stray 0-byte binary floating around elsewhere on the filesystem. Consul-template would have been one of the last things installed before the image was generated, so.. lord knows if those binaries were really valid (sort of kicking myself for not taking a SHA sum). I haven't seen this problem since re-generating our base VMs. 

Feel free to close out, and sorry for taking up your time!
